tests/test_co2eq_parameters.py

```diff
@@ -0,0 +1,188 @@
+"""Validate the CO2eq parameters."""
+
+import datetime
+import json
+import numbers
+import re
+import unittest
+
+
+def get_possible_modes():
+    """Get the set of possible modes."""
+    modes = set()
+    with open("web/src/helpers/constants.js", encoding="utf-8") as file_:
+        # The call to `eval` is a hack to parse the `modeOrder` array from the
+        # JavaScript source file.
+        for mode in eval(re.search(r"^const modeOrder = (\[$.*?^]);$",
+                                   file_.read(),
+                                   flags=re.DOTALL|re.MULTILINE)
+                           .group(1)):
+            if mode.endswith(" storage"):
+                modes.update((mode.replace("storage", "charge"),
+                              mode.replace("storage", "discharge")))
+            else:
+                modes.add(mode)
+    return modes
+
+
+def parse_json_file(path):
+    """Parse a JSON file."""
+    with open(path, encoding="utf-8") as file_:
+        return json.load(file_)
+
+
+class CO2eqParametersAll(unittest.TestCase):
+    """A test case for the CO2eq parameters."""
+    modes = get_possible_modes()
+    parameters = parse_json_file("config/co2eq_parameters_all.json")
+
+    @classmethod
+    def check_power_origin_ratios(cls, callback1, callback2):
+        """Apply the callback to each ratio in the 'powerOriginRatios' objects.
+        """
+        # It is useful to keep `test_power_origin_modes_are_valid` and
```

Comment:
This comment is already out of date, so I'd better remove the first sentence and keep the second.